### DISCUSSION TITLE
How to bring grouped window into focus?

### DISCUSSION THREAD

**Discussion by aksdb** [hyprwm/Hyprland#1540] [Q&A]
With `hyprctl clients` I get all windows. No matter if they are in a group or not. Which is fine; this is what I want anyway.
If I issue `hyprctl dispatch focuswindow addr:<someaddr from the client list>`, I get hyprland zu switch to the appropriate workspace and focus the given window. However, if that window is in the back of a group, nothing happens.

What I expected was, that the group gets brought into focus and the given window to the top.

Is this not-yet-implemented, a bug, or am I doing it wrong?

Edit: I also tried `hyprctl --batch "dispatch focuswindow address:$addr; dispatch bringactivetotop"`, without success.

Edit 2: Changed "stacked" to "grouped".

**Comment by vaxerski** [comment]
wdym "in the back of a stack"?

**Reply by aksdb** [reply]
With `togglegroup` the windows end up "stacked". So I have one active window, and the other window(s) are in the back. I can switch through them using `changegroupactive`.

I am looking for a quick way to find & switch to a window, even if it happens to be in a group/stack and is therefore "hidden" behind other window(s).

**Reply by vaxerski** [reply]
oh, you mean grouped. Stacked windows are floating windows.

No other way than cycling.

**Reply by aksdb** [reply]
Should I create a feature request for this or do you think this would be completely out of scope?

Naively it seems to me that `focuswindow` should implicitly make sure it's also the active window within a group. But I have almost zero knowledge of the inner workings of hypr (and wayland) so it might seem much easier in my head than it actually is.

**Reply by vaxerski** [reply]
feel free to make an issue, but it has to wait for #770 regardless

**Reply by burhanbilal** [reply]
Any updates?
